integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/petstore/PetStoreTestCase.java

```diff
@@ -0,0 +1,230 @@
+    private static final String link = "petStoreBackend:pet-be-inst";
+    private WebDriver webDriver;
+    private WebDriverWait wait;
+    private User alice;
+    private User bob;
+    private Cart cart;
+    private PetAccessory[] itemsOfAlice;
+    private Order order;
+
+    @BeforeClass
+    public void setup() {
+        WebDriverManager.getInstance(CHROME).setup();
+        webDriver = new ChromeDriver(new ChromeOptions().setHeadless(false));
+        wait = new WebDriverWait(webDriver, 120);
+
+        this.alice = new User("Alice", "Sanchez", "No 60, Regent street, " +
+                "New York.", "alice39", "alice123", webDriver, wait);
+        this.bob = new User("Bob", "Dylan", "No 36, Mayfair street, " +
+                "Los Angeles.", "bob39", "bob123", webDriver, wait);
+
+        this.cart = new Cart(webDriver);
+        this.order = new Order(webDriver);
+
+        PetAccessory aliceItem1 = new PetAccessory("Pet carrier cage",  50.00, 12
+                , Constants.PET_STORE_XPATH_PET_CARRIER_CAGE, webDriver);
+        PetAccessory aliceItem2 = new PetAccessory("Bone shaped toy",  5.00, 15
+                , Constants.PET_STORE_XPATH_BONE_SHAPED_TOY, webDriver);
+        itemsOfAlice = new PetAccessory[]{aliceItem1, aliceItem2};
+    }
+
+    @Test
+    public void buildBackEnd() throws Exception {
+        build("pet-be.bal", Constants.CELL_ORG_NAME, backEndImageName, version,
+                Paths.get(CELLERY_SCENARIO_TEST_ROOT, "pet-store", "pet-be").toFile().getAbsolutePath());
+    }
+
+    @Test
+    public void runBackEnd() throws Exception {
+        run(Constants.CELL_ORG_NAME, backEndImageName, version, backEndInstanceName, 600);
+    }
+
+    @Test
+    public void buildFrontEnd() throws Exception {
+        build("pet-fe.bal", Constants.CELL_ORG_NAME, frontEndImageName, version,
+                Paths.get(CELLERY_SCENARIO_TEST_ROOT, "pet-store", "pet-fe").toFile().getAbsolutePath());
+    }
+
+    @Test
+    public void runFrontEnd() throws Exception {
+        run(Constants.CELL_ORG_NAME, frontEndImageName, version, frontEndInstanceName, link, false,
+                600);
+    }
+
+    @Test
+    public void invoke() {
+        webDriver.get(Constants.DEFAULT_PET_STORE_URL);
+        String petAccessoriesHeader = webDriver.findElement(By.cssSelector("H1")).getText();
+        validateWebPage(petAccessoriesHeader, Constants.PET_STORE_WEB_CONTENT, "Pet store web page content is not " +
+                "as expected");
+    }
+
+    /**
+     * This tests sign in for user Alice.
+     * @throws InterruptedException
+     */
+    @Test
+    public void signInAlice() throws InterruptedException {
+        this.signIn(this.alice);
+    }
+
+    /**
+     * This tests add to cart for user Alice.
+     * @throws InterruptedException
+     */
+    @Test
+    public void addToCartAlice() throws InterruptedException {
+        for (int i = 0; i < itemsOfAlice.length; i++) {
+            TimeUnit.SECONDS.sleep(10);
```

Comment:
Is it possible to use waitFor element without having thread.sleep? Please check http://www.qaautomated.com/2016/10/implicit-wait-explicit-wait-and-fluent.html

Reply:
This method was tried and since there was no change in test failures, this was replaced with a Thread.sleep. Please check [L52](https://github.com/wso2-cellery/cellery-automation-scripts/blob/46a690cd077e98aa66eee9a19544d202aff07dc6/integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/PetStoreTestCase.java#L52) and [L102](https://github.com/wso2-cellery/cellery-automation-scripts/blob/46a690cd077e98aa66eee9a19544d202aff07dc6/integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/PetStoreTestCase.java#L102) in commit 46a690c

Reply:
Updated and used wait for element in all but 2 places. However as @azinneera mentioned, some tests were failing in jenkins server without explicit thread.sleep in above mentioned 2 places.

Reply:
Added a comment.